packages/devtools_app/lib/src/shared/diagnostics/references.dart

```diff
@@ -279,18 +282,26 @@ void _addLiveReferenceToNode(
   HeapObjectSelection heapSelection, {
   String namePrefix = '',
 }) {
-  if (instance is! InstanceRef && instance is! ContextRef) return;
+  if (instance is! ObjRef) throw 'Unexpected type: ${instance.runtimeType}.';
```

Comment:
Don't throw bare strings. Consider throwing `TypeError` or `ArgumentError` instead.

Reply:
Also, if this is always expected to be an `ObjRef`, the `instance` parameter should have type `ObjRef` or `ObjRef?`. We should avoid using `Object` when we have a known expected type as it's always better to have a static analysis failure than a runtime failure.

Reply:
I agree. I am not confident about expectations for this method. I just never saw non ObjRef so far. 
So, If I ever get non ObjRef, I (or user) will get exception and will add processing for it.

Reply:
I dream to refactor this code eventually to make contracts clearer.